internal/k8s/reference_checkers.go

```diff
@@ -121,6 +123,9 @@ func (rc *serviceReferenceChecker) IsReferencedByVirtualServer(svcNamespace stri
 	}
 
 	for _, u := range vs.Spec.Upstreams {
+		if rc.hasClusterIP && u.UseClusterIP {
```

Comment:
there is an edge case:
```yaml
  upstreams:
  - name: tea-1
    service: tea-svc
    has-cluster-ip: true
    port: 80
  - name: tea-2
    service: tea-svc
    port: 80
```    

in this case, IsReferencedByVirtualServer and IsReferencedByVirtualServerRoute need to return true.

So we need `continue` rather then `return false` in case of `if rc.hasClusterIP && u.UseClusterIP`